lib/api/wasm_c.cpp

```diff
@@ -157,6 +158,22 @@ wasm_valtype_t wasm_valtype_funcref(wasm::ValKind::FUNCREF);
     wasm::own<Name> ptr(obj);                                                  \
   }
 
+#define WASM_DECLARE_VEC_BASE_IMPL(name, Name)                                 \
+  WASMEDGE_CAPI_EXPORT void wasm_##name##_vec_new_uninitialized(               \
+      wasm_##name##_vec_t *out, size_t size) {                                 \
+    wasm::vec<Name> v1 = wasm::vec<Name>::make_uninitialized(size);            \
+    *out = {v1.size(), reinterpret_cast<wasm_val_t *>(v1.release())};          \
+  }                                                                            \
+  WASMEDGE_CAPI_EXPORT void wasm_##name##_vec_new_empty(                       \
+      wasm_##name##_vec_t *out) {                                              \
+    wasm_##name##_vec_new_uninitialized(out, 0);                               \
+  }                                                                            \
+                                                                               \
+  WASMEDGE_CAPI_EXPORT void wasm_##name##_vec_delete(wasm_##name##_vec_t *v) { \
```

Comment:
If this function is the same as `wasm_##name##_vec_delete` below, please follow the coding style.